core/src/main/java/org/kiwix/kiwixmobile/core/utils/SharedPreferenceUtil.kt

```diff
@@ -0,0 +1,173 @@
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+package org.kiwix.kiwixmobile.core.utils
+
+import android.content.Context
+import android.content.SharedPreferences
+import android.preference.PreferenceManager
+import androidx.appcompat.app.AppCompatDelegate
+import androidx.core.content.ContextCompat.getExternalFilesDirs
+import androidx.core.content.edit
+import io.reactivex.Flowable
+import io.reactivex.processors.PublishProcessor
+import org.kiwix.kiwixmobile.core.CoreApp.Companion.instance
+import org.kiwix.kiwixmobile.core.NightModeConfig
+import org.kiwix.kiwixmobile.core.NightModeConfig.Mode.Companion.from
+import java.io.File
+import java.util.Locale
+import javax.inject.Inject
+import javax.inject.Singleton
+
+/**
+ * Manager for the Default Shared Preferences of the application.
+ */
+@Singleton
+class SharedPreferenceUtil @Inject constructor(context: Context) {
+  private val sharedPreferences: SharedPreferences =
+    PreferenceManager.getDefaultSharedPreferences(context)
+  private val _prefStorages = PublishProcessor.create<String>()
+  val prefStorages = _prefStorages.startWith { prefStorage }
+  private val textZooms = PublishProcessor.create<Int>()
+  private val nightModes = PublishProcessor.create<NightModeConfig.Mode>()
+
+  val prefWifiOnly: Boolean
+    get() = sharedPreferences.getBoolean(PREF_WIFI_ONLY, true)
+
+  val prefHideToolbar: Boolean
+    get() = sharedPreferences.getBoolean(PREF_HIDE_TOOLBAR, true)
+
+  val prefIsFirstRun: Boolean
+    get() = sharedPreferences.getBoolean(PREF_IS_FIRST_RUN, true)
+
+  val prefFullScreen: Boolean
+    get() = sharedPreferences.getBoolean(PREF_FULLSCREEN, false)
+
+  val prefBackToTop: Boolean
+    get() = sharedPreferences.getBoolean(PREF_BACK_TO_TOP, false)
+
+  val prefNewTabBackground: Boolean
+    get() = sharedPreferences.getBoolean(PREF_NEW_TAB_BACKGROUND, false)
+
+  val prefExternalLinkPopup: Boolean
+    get() = sharedPreferences.getBoolean(PREF_EXTERNAL_LINK_POPUP, true)
+
+  val prefLanguage: String
+    get() = sharedPreferences.getString(PREF_LANG, "") ?: Locale.ROOT.toString()
+
+  val prefStorage: String
+    get() {
+      val storage = sharedPreferences.getString(PREF_STORAGE, null)
+      return when {
+        storage == null -> defaultStorage().also(::putPrefStorage)
+        !File(storage).exists() -> defaultStorage()
+        else -> storage
+      }
+    }
+
+  private fun defaultStorage(): String =
+    getExternalFilesDirs(instance, null)[0]?.path ?: instance.filesDir.path
```

Comment:
a comment was removed from here as compared to java. This code is a bit surprising so I would re-add that comment

Reply:
afaik, it was: "A workaround for emulators"